pandas/core/dtypes/cast.py

```diff
@@ -2096,6 +2096,9 @@ def maybe_cast_to_integer_array(
         )
         return casted
 
+    if all(isinstance(i, (int, np.integer)) for i in casted):
```

Comment:
you can `infer_dtype(casted) == 'integer'` i think

Reply:
Done